### PR TITLE
Views

### PR DESCRIPTION
**Pull request by rex-9**
- Implemented the design from the sneak peek wireframes.
    - Styling is nice to have but focused on making sure everything functions first.
    
- All Users page wireframe
- Single User page wireframe
- All Posts by a given User page wireframe
- Single Post page wireframe
are implemented here.

- Used methods that I have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- In order to see some posts or comments, I have added some data to my database with db/seed.rb file

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,7 @@
-<h1>Here is the list of users</h1>
\ No newline at end of file
+<div class="list">
+  <% @users.each do |user| %>
+    <%= link_to user do %>
+      <%= render partial: "user", object: user %>
+    <% end %>
+  <% end %>
+</div>
```
Comment (cwaku) [inline app/views/users/index.html.erb]: - [x] Kudos on making use of partials here. 💯

File: app/views/users/_user.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<article class="media is-align-items-center">
+  <div class="media-left">
+    <figure class="image is-128x128">
+    <%= image_tag user.photo %>
+    </figure>
+  </div>
+  <div class="media-content notification">
+    <div class="content">
+      <h4 class="title is-4"><%= user.name %></h4>
+      <p class="has-text-right is-size-7"><%= t('Posts_Counter', count: user.posts_counter)%></p>
```
Comment (cwaku) [inline app/views/users/_user.html.erb:10]: - [ ] The number of posts a user has isn't displaying on the view but rather being set as an HTML attribute. Kindly fix that.

View | HTML inspector
--|--
image | image

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,9 @@
-<h1>Here is the profile of a given user</h1>
\ No newline at end of file
+<%= render @user %>
+<% if @user.bio.present? %>
+	<div class="mt-5 notification">
+	  <h5 class="title is-6">Bio</h5>
+	  <p class="is-6"><%= @user.bio %></p>
+	</div>
+<% end %>
+<%= render partial: "posts/post", collection: @posts %>
```
Comment (cwaku) [inline app/views/users/show.html.erb]: - [ ] As part of the requirements, the list of the 3 most recent post are to be displayed here. Your current implementation doesn't have that. I suggest that you fix it.

Your Design | Template Design
--|--
image | image

File: app/views/posts/_post.html.erb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  <div class="mb-2">
+    <p><%= t('post', id: post.id)%></p>
+    <p class="is-size-7 is-italic"><%= t('posted_on', date: post.created_at.to_s(:nice_date)) %></p>
+  </div>
+  <%= link_to user_post_path(post.author.id, post.id) do %>
+    <h3 class="title"><%= post.title %></h3>
+  <% end %>
+  <p class="has-text-right is-size-7">
+    <%= t('comments_counter', count: post.comments_counter || 0 ) %>, <%= t('likes_counter', count: post.likes_counter || 0) %>
+  </p>
```
Comment (cwaku) [inline app/views/posts/_post.html.erb:11]: - [ ] Also fix here the issues of the counts being set HTML attributes, so it displays on the view. ( There are other occurrences in your project). Kindly fix them so the counts display correctly.

File: app/views/posts/_post.html.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<article class="notification my-2">
+  <div class="mb-2">
+    <p><%= t('post', id: post.id)%></p>
+    <p class="is-size-7 is-italic"><%= t('posted_on', date: post.created_at.to_s(:nice_date)) %></p>
+  </div>
+  <%= link_to user_post_path(post.author.id, post.id) do %>
+    <h3 class="title"><%= post.title %></h3>
+  <% end %>
```
Comment (cwaku) [inline app/views/posts/_post.html.erb:8]: - [ ] Post show page should display the post with all comments listed as shown on the wireframe. The following error occurs when a post is clicked. I recommend that you fix that.
image

**Review by cwaku — CHANGES_REQUESTED** [review]
Hi @rex-9 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!
There are a few issues 🐛  that you still need to work 🛠️  on to go to the next project ⏭️, but you are almost there! 🏃🏾 

## Highlights
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 

## Required Changes ♻️
It's ok to have bugs in your code. Please, see this code review as a chance to learn practical coding skills and new tips 🌟  that can improve your code quality 💯 and help you get a better job 💼.

I will guide 📔 you through fixing 🛠️ these bugs, follow my comments down below and make sure you don't miss any of them to get your project approved ✅

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @rex-9, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**